### PR TITLE
New bidder adapter  - Adquery

### PR DESCRIPTION
**Pull request by adquery**
## Type of change
- [x] New bidder adapter 

## Description of change
New bidder adapter

- test parameters for validating bids
```
{
  bidder: 'adquery',
  params: {
                placementId: '6d93f2a0e5f0fe2cc3a6e9e3ade964b43b07f897',
                type: 'banner300x250'
        }
}
```

### CODE REVIEW

File: modules/adqueryBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+      qid = storage.getDataFromLocalStorage('qid');
+    } else {
+      storage.setDataInLocalStorage('qid', qid);
+    }
+    let bid = bidRequests[0];
+    let request = {
+      placementCode: bid.placementId,
+      auctionId: bid.auctionId,
+      qid: qid,
+      userAgent: ua,
```
Comment (FilipStamenkovic) [inline modules/adqueryBidAdapter.js:52]: Do you need to send `userAgent` in the body? You can always read it from http headers.

File: modules/adqueryBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    const ua = navigator.userAgent;
+    let qid = Math.random().toString(36).substring(2) + Date.now().toString(36);
+    if (storage.getDataFromLocalStorage('qid')) {
+      qid = storage.getDataFromLocalStorage('qid');
+    } else {
+      storage.setDataInLocalStorage('qid', qid);
+    }
+    let bid = bidRequests[0];
+    let request = {
+      placementCode: bid.placementId,
```
Comment (FilipStamenkovic) [inline modules/adqueryBidAdapter.js:49]: You mean `bid.params.placementId` not `bid.placementId`?

File: modules/adqueryBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+      auctionId: bid.auctionId,
+      qid: qid,
+      userAgent: ua,
+      type: bid.type,
```
Comment (FilipStamenkovic) [inline modules/adqueryBidAdapter.js:53]: You mean `bid.params.type` not `bid.type`?

File: modules/adqueryBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    utils.logInfo(ADQUERY_LOG_PREFIX + 'onTimeout', timeoutData);
+    let params = {
+      bidder: timeoutData.bidder,
+      bId: timeoutData.bidId,
+      adUnitCode: timeoutData.adUnitCode,
+      timeout: timeoutData.timeout,
+      auctionId: timeoutData.auctionId,
+    };
+    let adqueryRequestUrl = utils.buildUrl({
+      protocol: 'https',
```
Comment (FilipStamenkovic) [inline modules/adqueryBidAdapter.js:129]: You already have `ADQUERY_BIDDER_DOMAIN_PROTOCOL`, why don't you use it here?

File: modules/adqueryBidAdapter.js [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+      bid: bid.adId,
+      mediaType: bid.mediaType,
+      cpm: bid.cpm,
+      requestId: bid.requestId,
+      adUnitCode: bid.adUnitCode,
+      adserverTargeting: getNestedParam(bid.adserverTargeting),
+    };
+
+    let adqueryRequestUrl = utils.buildUrl({
+      protocol: 'https',
```
Comment (FilipStamenkovic) [inline modules/adqueryBidAdapter.js:178]: Same as previous comment, you have `ADQUERY_BIDDER_DOMAIN_PROTOCOL`.

File: modules/adqueryBidAdapter.js [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+   */
+  buildRequests: (bidRequests, bidderRequest) => {
+    const ua = navigator.userAgent;
+    let qid = Math.random().toString(36).substring(2) + Date.now().toString(36);
+    if (storage.getDataFromLocalStorage('qid')) {
+      qid = storage.getDataFromLocalStorage('qid');
+    } else {
+      storage.setDataInLocalStorage('qid', qid);
+    }
+    let bid = bidRequests[0];
```
Comment (FilipStamenkovic) [inline modules/adqueryBidAdapter.js:47]: Why bidding only on 1st adUnit?  
If user requests bids for multiple adUnits at the same time, your bid adapter will send only bid request for 1 adUnit.  

If your backend, at the moment, doesn't support multi bid then in this function you should return array of bid request objects, not just single object. Then pbjs will send separate http request to your backed for each adUnit.

File: modules/adqueryBidAdapter.js [hunk 7]
```diff
@@ -0,0 +1,215 @@
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {BANNER} from '../src/mediaTypes.js';
+import * as utils from '../src/utils.js';
+import { getStorageManager } from '../src/storageManager.js';
+export const ADAPTER_VERSION = 1;
```
Comment (FilipStamenkovic) [inline modules/adqueryBidAdapter.js:5]: Where do you use this const?

**Review by FilipStamenkovic — CHANGES_REQUESTED** [review]
Take a look at my comments.  

Also, your test coverage is low (around 60%), you should increase it to be at least 80%.

File: modules/adqueryBidAdapter.js [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+
+  /** f
+   * @param {object} bid
+   * @return {boolean}
+   */
+  isBidRequestValid: (bid) => {
+    if (!(bid && bid.params && (bid.params.placementId))) {
+      return false;
+    }
+    return true;
```
[the remark this answers is not in the document]
Comment (adquery) [reply modules/adqueryBidAdapter.js]: ok thx

File: modules/adqueryBidAdapter.js [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+import * as utils from '../src/utils.js';
+import { getStorageManager } from '../src/storageManager.js';
+export const ADAPTER_VERSION = 1;
+
+const ADQUERY_GVLID = 902;
+const ADQUERY_BIDDER_CODE = 'adquery';
+const ADQUERY_BIDDER_DOMAIN_PROTOCOL = 'https';
+const ADQUERY_BIDDER_DOMAIN = 'bidder.adquery.io';
+const ADQUERY_USER_SYNC_DOMAIN = ADQUERY_BIDDER_DOMAIN_PROTOCOL + '://' + ADQUERY_BIDDER_DOMAIN + '/prebid/userSync?1=1';
+const ADQUERY_LOG_PREFIX = 'Adquery: ';
```
[the remark this answers is not in the document]
Comment (adquery) [reply modules/adqueryBidAdapter.js]: I removed

File: modules/adqueryBidAdapter.js (hunk 1 above)
Comment (adquery) [reply modules/adqueryBidAdapter.js]: you are right I deleted  this property

File: modules/adqueryBidAdapter.js (hunk 5 above)
Comment (adquery) [reply modules/adqueryBidAdapter.js]: I fixed

File: modules/adqueryBidAdapter.js (hunk 6 above)
Comment (adquery) [reply modules/adqueryBidAdapter.js]: I fixed

File: modules/adqueryBidAdapter.js (hunk 7 above)
Comment (adquery) [reply modules/adqueryBidAdapter.js]: unused I removed

File: modules/adqueryBidAdapter.js (hunk 3 above)
Comment (adquery) [reply modules/adqueryBidAdapter.js]: it is correct bid.type

File: modules/adqueryBidAdapter.js (hunk 2 above)
Comment (adquery) [reply modules/adqueryBidAdapter.js]: it is correct bid.placementId

**Comment by adquery** [comment]
I increase test coverage to about 85 percent. Thank you for review

**Comment by MartianTribe** [comment]
Related docs PR: https://github.com/prebid/prebid.github.io/pull/3275

**Review by FilipStamenkovic — CHANGES_REQUESTED** [review]
Your bidder is not working properly.  

I defined 2 adUnits with your bidder with following params:

```
bids: [{
        bidder: 'adquery',
        params: {
            placementId: '6d93f2a0e5f0fe2cc3a6e9e3ade964b43b07f897',
            type: 'banner300x250'
        }
 }]
```  
and
```
bids: [{
        bidder: 'adquery',
        params: {
            placementId: '123',
            type: 'banner300x250'
        }
    }]
```

These params are not sent in your http request. Furthermore, test responses you are sending are not properly matched with bid request.

File: modules/adqueryBidAdapter.js [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+
+  /**
+   * @param {*} response
+   * @param {ServerRequest} request
+   * @return {Bid[]}
+   */
+  interpretResponse: (response, request) => {
+    utils.logInfo(response);
+    utils.logInfo(request);
+    const res = response && response.data;
```
Comment (FilipStamenkovic) [inline modules/adqueryBidAdapter.js:61]: Isn't it `response.body.data`?

Comment (adquery) [reply modules/adqueryBidAdapter.js]: I fixed

**Comment by adquery** [comment]
I fixed multiple adUnits

**Review by FilipStamenkovic — APPROVED** [review]

**Pull request merged**